packages/thirdweb/src/react/native/ui/connect/ConnectedButton.tsx

```diff
@@ -0,0 +1,184 @@
+      account,
+      props.detailsButton?.displayBalanceToken,
+    );
+  return (
+    wallet &&
+    account && (
+      <ThemedButton
+        theme={theme}
+        onPress={() => {
+          props.onClose();
+          disconnect(wallet);
+        }}
+        style={{
+          paddingHorizontal: spacing.md,
+          paddingVertical: spacing.smd,
+        }}
+      >
+        <View style={styles.row}>
+          <WalletImageOrEns
+            wallet={wallet}
+            account={account}
+            ensAvatar={ensAvatarQuery.data}
+          />
+          <View style={styles.col}>
+            <ThemedText
+              theme={theme}
+              type="defaultSemiBold"
+              style={{
+                color: theme.colors.primaryButtonText,
+                fontSize: fontSize.sm,
+              }}
+            >
+              {addressOrENS}
+            </ThemedText>
+            <ThemedText
+              theme={theme}
+              type="subtext"
+              style={{
+                fontSize: fontSize.sm,
+              }}
+            >
+              {balanceQuery.data
+                ? Number(balanceQuery.data.displayValue).toFixed(3)
+                : "-"}{" "}
+              {balanceQuery.data?.symbol}{" "}
+            </ThemedText>
+          </View>
+        </View>
+        {/* <ThemedText
+          theme={theme}
+          type="defaultSemiBold"
+          style={{ color: theme.colors.primaryButtonText }}
+        >
+          Disconnect
+        </ThemedText> */}
+      </ThemedButton>
+    )
+  );
+}
+
+const WalletImageOrEns = (props: {
+  wallet: Wallet;
+  account: Account;
+  ensAvatar?: string | null;
+}) => {
+  const { wallet, account, ensAvatar } = props;
+  const { data: imageData } = useQuery({
+    queryKey: ["wallet-image-or-ens", wallet.id, account.address],
+    queryFn: async () => {
+      let imageData: string;
+      let imageType: "xml" | "image" | "url";
+      if (ensAvatar) {
+        return {
+          data: ensAvatar,
+          type: "url",
+        };
+      }
+      if (wallet.id === "inApp") {
+        const lastAuthProvider = await getLastAuthProvider(nativeLocalStorage);
+        imageType = "xml";
+        switch (lastAuthProvider) {
+          case "phone":
+            imageData = PHONE_ICON;
+            break;
+          case "email":
+            imageData = EMAIL_ICON;
+            break;
+          case "google":
+            imageData = GOOGLE_ICON;
+            break;
+          case "apple":
+            imageData = APPLE_ICON;
+            break;
+          case "facebook":
+            imageData = FACEBOOK_ICON;
+            break;
+          default:
+            imageData = WALLET_ICON;
+            break;
+        }
+        return {
+          data: imageData,
+          type: imageType,
+        };
+      }
+      const externalWalletImage = await getWalletInfo(wallet.id, true);
+      return {
+        data: externalWalletImage,
+        type: "image",
```

Comment:
Not sure if this will break things but an ecosystem wallet will return an image url rather than a base64 string from `getWalletInfo`

Reply:
I changed this in the next PR. Now accepts any kind of image uri